Fix problem with read_pubkey_file error handling.

BobWall23 opened the pull request:
Read_pubkey_file was returning 0 even if it couldn't read the key file.
Fixed to return proper error and success values.
Also removed some crusty test / experimental code from iron-gpg.c

BobWall23 committed:
Fix problem with read_pubkey_file error handling.

Read_pubkey_file was returning 0 even if it couldn't read the key file.
Fixed to return proper error and success values.
Also removed some crusty test / experimental code from iron-gpg.c

BobWall23 commented:
Fixes Issue #15

coltfred commented on iron-gpg.c:
```diff
@@ -3999,14 +3999,25 @@ static int
-	int retval = 0;
+	int retval = -1;
+
+	if (rsa_key	!= NULL) {
```
Why the funky amount of space in here?

BobWall23 replied:
F'ing tabs vs. spaces.

BobWall23 committed:
Fixed phantom tab

BobWall23 committed:
Code cleanup

Macros can be your friend! Squashed down some repetitive code into less
obnoxiousness.

zmre commented:
Again, have seen this same diff in two other PRs. Am using an iPad app right now so I guess it could be having problems?
